infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -346,7 +347,18 @@ static void restartServer(boolean resetData) {
     private static void cleanShutdown() {
         log.info("CleanShutdown: Starting Cleanup.");
         shutdownServer = true;
-        activeServer.close();
+        try {
+            CorfuServerNode current = activeServer;
```

Comment:
what about - replace `volatile` by AtomicReference? Then your code will be cleaner, no need for checking for null and bringing additional mental complexity to the code?

Reply:
Ok, looks like AtomicReference doesn't have good API for that. Nevermind

Reply:
Ok.